src/format.ts

```diff
@@ -15,30 +20,30 @@ function replaceHighlightTags(value, highlightPreTag, highlightPostTag) {
   return newHighlightValue.toString()
 }
 
-function createHighlighResult({
+function createHighlighResult<T extends Record<string, any>>({
   formattedHit,
   highlightPreTag,
   highlightPostTag,
-}) {
+}: { formattedHit: T } & Params) {
```

Comment:
Why is it both `{ formattedHit: T }` and `Params` as return types?

Reply:
Because formattedHit is not defined in Params.

If you remove it, line 24 will be marked as error:
> Property 'formattedHit' does not exist on type '{ page?: number | undefined; hitsPerPage?: number | undefined; highlightPreTag?: string | undefined; highlightPostTag?: string | undefined; snippetEllipsisText?: string | undefined; attributesToSnippet?: string[] | undefined; query: string; facets: string[]; } & SearchParams<...>'.ts(2339)

However, it would probably be a good idea to refactor this into a separate named type. One of the rough edges I mentioned in the initial description 😉